internal/test/e2e/e2e_test.go

```diff
@@ -181,51 +181,114 @@ func setupLocalCluster(t *testing.T) (*cluster.Context, func()) {
+		defer cancel()
+
+		etcdClient, err := etcd.New(etcdConfig)
+		require.NoError(t, err)
+
+		err = try.Eventually(func() error {
+			members, err := etcd.NewMembersAPI(etcdClient).List(ctx)
+			if err != nil {
+				return err
+			}
+
+			if len(members) != expectedClusterSize {
+				return errors.New(fmt.Sprintf("expected %d etcd peers, got %d", expectedClusterSize, len(members)))
+			}
+			return nil
+		}, time.Minute*2, time.Second*5)
+		require.NoError(t, err)
+	})
+
+	t.Run("EtcdClusterStatus", func(t *testing.T) {
+		t.Parallel()
+		err := try.Eventually(func() error {
+			t.Log("")
+			members, err := kubectl.Get("--namespace", "default", "etcdcluster", "my-cluster", "-o=jsonpath='{.status.members...name}'")
+
+			if err != nil {
+				return err
+			}
+			// Don't assert on exact members, just that we have three of them.
+			if len(strings.Split(members, " ")) != 3 {
+				return errors.New(fmt.Sprintf("Expected etcd member list to have three members. Had %d.", len(members)))
+			}
+			return nil
+		}, time.Minute*2, time.Second*5)
+		require.NoError(t, err)
+	})
+}
+
+func persistenceTests(t *testing.T, kubectl *kubectlContext) {
+	err := kubectl.Apply(
+		"--filename", "testdata/persistence_tests/",
 	)
 	require.NoError(t, err)
 
-	etcdClient, err := etcd.New(etcdConfig)
-	require.NoError(t, err)
+	expectedValue := "foobarbaz"
+
+	etcdctl := func(args ...string) (string, error) {
+		return kubectl.Run(
+			append(
+				[]string{
+					"--quiet",
+					"--restart=Never",
+					"--rm",
+					"--image=quay.io/coreos/etcd:v3.2.27",
+					"--attach",
+					"etcdctl",
+					"--",
+					"etcdctl",
+					"--insecure-discovery",
+					"--discovery-srv=cluster1",
+				}, args...,
+			)...,
+		)
+	}
 
-	err = try.Eventually(func() error {
-		t.Log("Checking if ETCD is available")
-		members, err := etcd.NewMembersAPI(etcdClient).List(ctx)
-		if err != nil {
-			return err
-		}
+	var out string
 
-		if len(members) != expectedClusterSize {
-			return errors.New(fmt.Sprintf("expected %d etcd peers, got %d", expectedClusterSize, len(members)))
-		}
-		return nil
-	}, time.Minute*2, time.Second*10)
-	require.NoError(t, err)
-	t.Log("ETCD is reachable from host machine")
+	err = try.Eventually(func() (err error) {
+		out, err = etcdctl("set", "--", "foo", expectedValue)
+		return err
+	}, time.Minute*2, time.Second*5)
 
-	err = try.Eventually(func() error {
-		t.Log("Checking if the EtcdCluster resource has the correct status")
-		members, err := kubectl.Get("--namespace", "default", "etcdcluster", "my-cluster", "-o=jsonpath='{.status.members...name}'")
+	require.NoError(t, err, out)
 
-		if err != nil {
-			return err
-		}
-		// Don't assert on exact memebers, just that we have three of them.
-		if len(strings.Split(members, " ")) != 3 {
-			return errors.New(fmt.Sprintf("Expected etcd member list to have three members. Had %d.", len(members)))
-		}
-		return nil
-	}, time.Minute*2, time.Second*10)
+	err = kubectl.Delete(
+		"pods",
```

Comment:
Can we delete the entire EtcdCluster resource instead/as well in another test? We should be able to recover from complete deletion of the EtcdCluster resource, and when recreated with existing PVCs the cluster should come online still 😄

Reply:
Done.